x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/es_deprecations_table.tsx

```diff
@@ -146,6 +156,13 @@ export const EsDeprecationsTable: React.FunctionComponent<Props> = ({
   deprecations = [],
   reload,
 }) => {
+  const {
+    services: { api },
+  } = useAppContext();
+
+  const { data } = api.useLoadMlUpgradeMode();
```

Comment:
What are your thoughts on fetching this within the flyout? If I'm not mistaken, this request will be made every time the ES deprecations page is loaded, even if there are no ML deprecations.

Also, is there anything we have to consider if the user has a `basic` license, or should this still work as expected? I also see there are [privileges](https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-set-upgrade-mode.html#ml-set-upgrade-mode-prereqs) associated with the API.

Reply:
I figured that if we fetch within the flyout we have two choices:

* We actually make the fetch when the flyout opens up, which will cause layout shifting due to callouts being rendered above the deprecation details
* Fetch the status in the `GET ml_snapshots/{jobId}/{snapshotId}` endpoint (my first implementation was based on this one), but that would generate an extra es api call per request. This one could work nicely and I don't mind changing the implementation to accommodate, but just wasn't sure how much we try to optimize for es api calls or if it's fast enough that it doesn't make much of a difference.

I don't think theres anything special we should do for `basic` license, I've done most of my testing using that one and it works just fine. 

Good catch in regards of the privileges! I don't think its a problem though, the only api we call is `GET _ml/info` for which the user should have the `monitor_ml` cluster privilege. At the moment if a user doesn't have it, the kibana api `GET upgrade_assistant/ml_upgrade_mode` will fail and the callout won't be rendered but actions will. Upon performing an action they will get an error accordingly. If we don't want it to fail, we could check for `monitor_ml` privileges before calling `GET _ml/info` and assume its disabled if we don't have privileges. WDYT?

Reply:
Thanks for the explanation!

I understand the concerns. It just felt a little strange to me to make a request that may or may not be necessary. I think there are two conditions that would need to be met for it to be used: (1) user needs to have at least 1 ML deprecation and (2) user needs to open the ML deprecation flyout. 

That said, I’m happy to leave as-is for now and see if the current approach causes any issues.

Reply:
As far as the second part -

> I don't think theres anything special we should do for basic license, I've done most of my testing using that one and it works just fine.

👍 Agreed - `GET _ml/info` seems to work fine. It looks like ES will return an error if a user for some reason has a basic license but then tries to upgrade/delete a snapshot (probably an edge case).

<img width="268" alt="Screen Shot 2021-09-23 at 10 06 28 AM" src="https://user-images.githubusercontent.com/5226211/134523528-2a7de145-df38-42f8-af09-15f81c25a8a0.png">

> Good catch in regards of the privileges! I don't think its a problem though, the only api we call is GET _ml/info for which the user should have the monitor_ml cluster privilege. At the moment if a user doesn't have it, the kibana api GET upgrade_assistant/ml_upgrade_mode will fail and the callout won't be rendered but actions will. Upon performing an action they will get an error accordingly. If we don't want it to fail, we could check for monitor_ml privileges before calling GET _ml/info and assume its disabled if we don't have privileges. WDYT?

👍 I think this is outside the scope of this PR, but I think it would be a nice-to-have to check privileges for the ML action, as I just noticed the reindexing action does show a callout if the user does not have correct privileges.

<img width="513" alt="Screen Shot 2021-09-23 at 10 06 12 AM" src="https://user-images.githubusercontent.com/5226211/134524220-ee4f9b42-0ed5-4c40-8711-70a16bd37b07.png">